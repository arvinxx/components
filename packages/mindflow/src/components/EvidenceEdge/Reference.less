@import '../../themes/theme';

.mind-edge-references {
  color: @text-color-secondary;

  &-dot {
    width: 4px;
    height: 4px;
    margin-right: 8px;
    margin-left: 8px;
    background: @text-color-third;
    border-radius: 100%;
  }

  &-item {
    display: flex;
    align-items: center;
    margin-bottom: 12px;
    color: @text-color-secondary;
    font-size: 14px;
  }

  &-tag {
    margin-left: 4px;
    padding: 0 4px;
  }

  &-foot {
    display: flex;
    flex-direction: row-reverse;
  }
}
.ant-popover {
  z-index: 2000;
  &-inner-content {
    padding: 8px;
    padding-top: 16px;
  }
}
