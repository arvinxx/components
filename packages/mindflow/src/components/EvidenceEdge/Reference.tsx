import type { FC } from 'react';
import React from 'react';
import { Button, Divider, Tag } from 'antd';

import type { EdgeReference } from '../../types';
import { mapReferenceTypeToColor } from '../../utils';

import './Reference.less';

interface EvidencePanelProps {
  references: EdgeReference[];
  setVisible: (visible: boolean) => void;
}

const EvidencePanel: FC<EvidencePanelProps> = ({ references, setVisible }) => {
  const mapTypeToText = (type: EdgeReference['type']) => {
    switch (type) {
      case 'ground':
        return '事实';
      case 'warrant':
        return '推测';

      case 'backing':
        return '支援';
      case 'rebuttal':
        return '反驳';
      default:
        return '未定';
    }
  };

  const Content: FC<{ reference: EdgeReference }> = ({ reference }) => {
    return (
      <>
        <div>
          <Tag
            className="mind-edge-references-tag"
            color={mapReferenceTypeToColor(reference.type)}
          >
            {mapTypeToText(reference.type)}
          </Tag>
        </div>
        <div className="mind-edge-references-text">{reference.title}</div>
      </>
    );
  };

  return (
    <div className="mind-edge-references">
      {references.map((ref) => {
        return !ref.url ? (
          <div className="mind-edge-references-item" key={ref.id}>
            <Content reference={ref} />
          </div>
        ) : (
          <a
            key={ref.id}
            href={ref.url}
            target={'_blank'}
            className="mind-edge-references-item"
          >
            <Content reference={ref} />
          </a>
        );
      })}
      <Divider dashed style={{ margin: '8px 0' }} />
      <div className="mind-edge-references-foot">
        <Button size={'small'} onClick={() => setVisible(false)}>
          关闭
        </Button>
      </div>
    </div>
  );
};

export default EvidencePanel;
