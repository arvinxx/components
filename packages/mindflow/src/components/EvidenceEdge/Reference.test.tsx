import React from 'react';
import { render } from '@testing-library/react';

import Reference from './Reference';

describe('Reference', () => {
  it('空档案', () => {
    const { container } = render(
      <Reference references={[]} setVisible={() => {}} />,
    );
    expect(container).toMatchSnapshot();
  });
  it('列表', () => {
    const { container } = render(
      <Reference
        references={[
          { id: '1', title: 'node1', type: 'rebuttal' },
          { id: '2', title: 'node2', url: 'https://url' },
          { id: '3', title: 'node3', type: 'backing' },
          { id: '4', title: 'node4', type: 'ground' },
          { id: '5', title: 'node5', type: 'warrant' },
        ]}
        setVisible={() => {}}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  it('点击关闭', async () => {
    const onClick = jest.fn();
    const { findByText } = render(
      <Reference references={[]} setVisible={onClick} />,
    );
    const btn = await findByText('关 闭');

    btn.click();

    expect(onClick).toBeCalledTimes(1);
  });
});
