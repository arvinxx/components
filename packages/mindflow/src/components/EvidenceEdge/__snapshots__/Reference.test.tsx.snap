// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reference 列表 1`] = `
<div>
  <div
    class="mind-edge-references"
  >
    <div
      class="mind-edge-references-item"
    >
      <div>
        <span
          class="ant-tag ant-tag-purple mind-edge-references-tag"
        >
          反驳
        </span>
      </div>
      <div
        class="mind-edge-references-text"
      >
        node1
      </div>
    </div>
    <a
      class="mind-edge-references-item"
      href="https://url"
      target="_blank"
    >
      <div>
        <span
          class="ant-tag ant-tag-has-color mind-edge-references-tag"
          style="background-color: gray;"
        >
          未定
        </span>
      </div>
      <div
        class="mind-edge-references-text"
      >
        node2
      </div>
    </a>
    <div
      class="mind-edge-references-item"
    >
      <div>
        <span
          class="ant-tag ant-tag-yellow mind-edge-references-tag"
        >
          支援
        </span>
      </div>
      <div
        class="mind-edge-references-text"
      >
        node3
      </div>
    </div>
    <div
      class="mind-edge-references-item"
    >
      <div>
        <span
          class="ant-tag ant-tag-green mind-edge-references-tag"
        >
          事实
        </span>
      </div>
      <div
        class="mind-edge-references-text"
      >
        node4
      </div>
    </div>
    <div
      class="mind-edge-references-item"
    >
      <div>
        <span
          class="ant-tag ant-tag-blue mind-edge-references-tag"
        >
          推测
        </span>
      </div>
      <div
        class="mind-edge-references-text"
      >
        node5
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal ant-divider-dashed"
      role="separator"
      style="margin: 8px 0px;"
    />
    <div
      class="mind-edge-references-foot"
    >
      <button
        class="ant-btn ant-btn-sm"
        type="button"
      >
        <span>
          关 闭
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`Reference 空档案 1`] = `
<div>
  <div
    class="mind-edge-references"
  >
    <div
      class="ant-divider ant-divider-horizontal ant-divider-dashed"
      role="separator"
      style="margin: 8px 0px;"
    />
    <div
      class="mind-edge-references-foot"
    >
      <button
        class="ant-btn ant-btn-sm"
        type="button"
      >
        <span>
          关 闭
        </span>
      </button>
    </div>
  </div>
</div>
`;
